File: src/main/java/com/web/model/dto/AttendanceDTO.java
package com.web.model.dto;

import com.web.enums.operation.SortDirection;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class AttendanceDTO {
    private Long id;
    private Timestamp checkInTime;
    private Timestamp checkOutTime;
    private String method;
    private MemberDTO memberEntity;
    private String sortFieldName;
    private SortDirection sortDirection;
}

File: src/main/resources/templates/admin/assets/js/model/dto/AttendanceDTO.js
import {MemberDTO} from "/templates/admin/assets/js/model/dto/MemberDTO.js";

export class AttendanceDTO {
    /**
     * AttendanceDTO constructor
     * @param id
     * @param {MemberDTO} MemberEntity
     * @param checkInTime
     * @param checkOutTime
     * @param method
     */
    constructor(id, MemberEntity, checkInTime, checkOutTime, method) {
        this.id = id;
        this.memberEntity = MemberEntity instanceof MemberDTO ? MemberEntity : new MemberDTO();
        this.checkInTime = checkInTime;
        this.checkOutTime = checkOutTime;
        this.method = method;
    }
}